### ISSUE TITLE
Uncaught (in promise) DOMException: The play() request was interrupted by a call to pause()

### ISSUE THREAD

**Issue by joseluisq** [sampotts/plyr#331]
When the video is stopped and I'm dragging the `plyr__progress` bar to the next minute, 
it appears this error:

`Plyr v1.8.12`

![image](https://cloud.githubusercontent.com/assets/1700322/17452652/e1bedc22-5b36-11e6-8698-d2cc5e5e65c7.png)

**Comment by SamPotts** [comment]
This seems to be a Chrome issue by the looks of things:
https://bugs.chromium.org/p/chromium/issues/detail?id=593273

**Comment by filoghost** [comment]
There is a workaround, not pretty but it should work: http://stackoverflow.com/questions/36803176/how-to-prevent-the-play-request-was-interrupted-by-a-call-to-pause-error

**Comment by SamPotts** [comment]
Closing this and waiting on a Chrome fix

**Comment by neutrosolutions** [comment]
From html video TAG remove autoplay
and use this code to play.
`hls.attachMedia(video);
                hls.on(Hls.Events.MANIFEST_PARSED, function () {
                    video.play();
                });
hls.on(Hls.Events.ERROR, function (event, data) {
                    if (data.fatal) {
                        switch (data.type) {
                            case Hls.ErrorTypes.NETWORK_ERROR:
                                // try to recover network error
                                console.log("fatal network error encountered, try to recover");
                                hls.startLoad();
                                break;
                            case Hls.ErrorTypes.MEDIA_ERROR:
                                console.log("fatal media error encountered, try to recover");
                                hls.recoverMediaError();
                                break;
                            default:
                                // cannot recover
                                hls.destroy();
                                break;
                        }
                    }
                });`

**Comment by alejandroiglesias** [comment]
I would like to reopen this because the aforementioned Chrome issue says the error is fixed for when calling `HTMLMediaElement.pause` before `HTMLMediaElement.play`, but calling `HTMLMediaElement.play` and then `HTMLMediaElement.pause` while the media is loading still throws the error.

```
// fixed
media.pause();
media.play();

// won't fix; still shows the issue.
media.play();
media.pause();
```

`HTMLMediaElement.play` is async and returns a promise. The error complains about having called `HTMLMediaElement.pause` before the playback started.

I would like to quote to the Chromium issue owner:
> This has been resolved a long time ago. If you are seeing an error but that your website otherwise behave exactly the same, this is not a bug in Chrome but in your website. If you are seeing an error in the console *and* your website stopped behaving properly, please file another bug with a test page.

In this case, everything behaves exactly the same and it won't get fixed by the Chromium team because the call to `HTMLMediaElement.play` returns a promise that gets resolved when the media element is ready to play. At that point, the call to `HTMLMediaElement.pause` can be made. Since we rely on Plyr to handle HTMLMediaElement, it should either know about the promise returned by `HTMLMediaElement.play` and smartly call `HTMLMediaElement.pause` when it's resolved, or at least provide a way to interact with the promise by returning it when calling `plyr.play`. I tried to look at `plyr.isPaused()` and `plyr.getMedia().paused` in order to work around it myself, but they're `false` even though they didn't start playing yet because of buffering. I also tried to use `plyr.getMedia().play()` instead of `plyr.play()` in order to get the promise and then always call pause/stop when the promise resolves, but eventually, I get the error again, so maybe I'm missing something. I think that Plyr could keep a track of this promise internally so that users don't need to worry about this issue.

**Comment by laukstein** [comment]
Chrome post about how to fix this issue https://developers.google.com/web/updates/2017/06/play-request-was-interrupted

**Comment by pionl** [comment]
Hi there, 
this can't be fixed at our end, this is problem when the source is still loading and pause/destroy is called (event if the plyr container is removed from dom - chrome indicates pause event).

Only solution is to catch promise in the play/pause (this can be only done in plyr source).
Martin.

**Comment by lichenbuliren** [comment]
I use pretty javascript code solve it:

``` javascript
video.src = 'xxx';
video.load();
setTimeout(function() {
    video.play();
}, 0);
```

**Comment by mhluska** [comment]
I'm not able to solve this problem when there are multiple video elements on the screen. The error seems to trigger when I call `.play()` on one while the previous one is still loading.

**Comment by ramyahk02** [comment]
Is there any update on this issue? I still get the error "Uncaught (in promise) DOMException: The play() request was interrupted by a call to pause()." Trying to use the Promise function to handle this gives 'playPromise.then is not a function' error. Any suggestion would help.

**Comment by sampotts** [comment]
No update but it can be safely ignored. v3 will return a `Promise` for calls to `.play()` and will be in beta very soon.

**Comment by liamtarpey** [comment]
I can still reproduce this when passing `autoplay: true` to the player config when mounting my component.

Something like this fixes it for me:

```
if (autoplayWanted) {
    player.play();
}
```

**Comment by clark-stevenson** [comment]
I can form with @liamtarpey and also thank you for the work around.

**Comment by intelligence** [comment]
Yup, same here. Get `Uncaught (in promise) DOMException` thrown in my face if I set `autoplay: true`. If I remove it it'll post no error.

**Comment by Hydra9268** [comment]
Can confirm that lichenbuliren's autoplay fix works for audio.

My code below:
`
var player = new Plyr('#player', { controls: [ 'restart', 'play', 'progress', 'mute', 'volume' ]});
player.source = {
	type: 'audio',
	sources: [{
		src: 'http://u7map-leaflet.s3.amazonaws.com/files/music/' + musicID + '.mp3',
		type: 'audio/mp3'
	}]
};
setTimeout(function() {
	player.play();
}, 0);
`

The 0ms timeout is required to give Plyr the time it needs to stop the current audio from play before loading in the new. You can see it the fix in action at http://kxmode.com/u7map/

**Comment by electroheadfx** [comment]
I have the same issue with the last version on safari mac. Work in Firefox and chrome. The issue is here: https://github.com/sampotts/plyr/issues/1385

**Comment by antongorodezkiy** [comment]
Have the same issue in Chromium 75 on Ubuntu

**Comment by laukstein** [comment]
Why https://github.com/sampotts/plyr isn't considering fixing this issue?

**Comment by Venryx** [comment]
Just wanted to mention that, for me @lichenbuliren's solution didn't work (the error still showed).

This worked for me instead:
```
video.pause();
video.addEventListener("canplay", function onCanPlay() {
	video.removeEventListener("canplay", onCanPlay);
	video.play();
});
video.load();
```

**Comment by karljakober** [comment]
Still encountering this issue coming from plyr...

**Comment by laukstein** [comment]
If according to @karljakober this issue still accrues on the latest Plyr, why this issue isn't reopen?

**Comment by laukstein** [comment]
@karljakober can you add screenshot of issue you notice? Does it post console info or error?
Ref https://github.com/pionl/plyr/commit/58677033cc9ec9c3aea1d77f0bd86b1b902034d9#diff-2999d34c7d48fa26668dd28082cd6303R2079

**Comment by Benny739** [comment]
For me it's a console.error. Happens when die video is buffering and you press pause.

**Comment by kush1960** [comment]
We're still experiencing the same issue using HLS.js 
Same console error as Benny739 and only happens when pausing a video while it's still buffering.

**Comment by dohahelmy** [comment]
issue still happening, and no solutions seems to work

**Comment by Hydra9268** [comment]
My code example above converted to a promise. 

```js
function initializePlayer(musicID) {
  return new Promise((resolve, reject) => {
    try {
      var player = new Plyr('#player', {
        controls: ['restart', 'play', 'progress', 'mute', 'volume']
      });

      player.source = {
        type: 'audio',
        sources: [{
          src: '[URL TO MP3 FILES]',
          type: 'audio/mp3'
        }]
      };

      // Wait for the player to be ready before playing
      player.once('ready', () => {
        player.play().then(() => {
          resolve(player);
        }).catch((error) => {
          reject(error);
        });
      });
    } catch (error) {
      reject(error);
    }
  });
}

// Usage
initializePlayer(musicID)
  .then((player) => {
    console.log('Player initialized and playing:', player);
  })
  .catch((error) => {
    console.error('Error initializing player:', error);
  });
```

This should resolve any interruption issues.
